packages/js/src/classic-editor/helpers/dom.js

```diff
@@ -188,6 +188,39 @@ export const getPostDate = () => {
 	], timezone ).toISOString();
 };
 
+/**
+ * Get the post category checkboxes elements from the document.
+ *
+ * @returns {HTMLInputElement[]} The category checkboxes.
+ */
+export const getPostCategoryCheckboxes = () => {
+	return [ ...document.querySelectorAll( "#categorychecklist input[type=checkbox]" ) ];
+};
+
+/**
+ * Gets the current post's categories from the document.
+ *
+ * @returns {{name: string, id: string}[]} The post's categories.
+ */
+export const getPostCategories = () => {
+	const checkboxes = getPostCategoryCheckboxes();
+
+	if ( checkboxes ) {
+		const checkedCheckboxes = checkboxes.filter( checkbox => checkbox.checked );
+		return checkedCheckboxes.map( checkbox => (
+			{
+				id: checkbox.value,
+				name: [ ...checkbox.parentElement.childNodes ]
+					.filter( node => node.nodeType === Node.TEXT_NODE )
+					.map( node => node.textContent )
+					.join( " " )
+					.trim(),
```

Comment:
Note for reviewers: this is a bit of a hacky solution.

We add the hidden text _"(Primary category)"_ behind the selected primary category within the classic editor for screen readers. This means that using the original approach (using `checkbox.parentElement.innerText`) would result in the string _"Category name (Primary category)"_.

This hacky approach makes sure that _"(Primary category)"_ does not occur in the category name.

Reply:
@hansjovis I've also discussed this with @mhkuu and we cannot think of a better solution than this 😄 . There is only this small thing that we actually don't need the space when using `.join( " " )` if we understand this correctly. because then the spaces will be trimmed in the next step anyway. We've tested with `.join( "" )` and it works the same. Is it okay if I make this change?

Reply:
I removed the join 🙂 .